Add jsonline output support to stdout logger

olksdr opened the pull request:
While it's nice for humans to see pretty log messages and all those
indentation it's much more difficult for machine consumption.

This introduces a change from using `encodePretty'` to simple `encode`
which will produce jsonline for each log message.

In modern word of containers and K8s it is the required way for logging.

An example of how the new logging output will look like:


```json
{"component":"UDP","data":{"from":"127.0.0.1:37392","answer":"1.2.1.1","server":"ec2121e7bdd4","question":"1.2.1.1.example.com."},"insertion_time":"","message":"","timestamp":"2020-07-18 10:07:27","level":"INFO"}
```

arybczak commented:
Thanks for the PR. I'm fine with adding that, but it should be optional/configurable.

olksdr commented:
@arybczak , thanks for looking into this!  
you're definitely correct. I will update this PR to have a different stdout logger for jsonlines 😄

arybczak commented:
Yes please, making it a separate logger would be best :pray:

olksdr commented:
hey @arybczak, 

I've updated the PR where:
* added another logger for dumping jsonlines to `stdout` 
* buffers will be flushed after write, which is quite important too
* the previous loggers are look the same to keep the backwards compatibility with existing software
 
Could you, please, take another look?

arybczak commented on log-base/src/Log/Data.hs:
```diff
@@ -90,6 +91,24 @@ showLogMessage mInsertionTime LogMessage{..} = T.concat $ [
     }
 
+-- | Render a 'LogMessage' to 'Text' as a JsonLine.
+showJsonLineLogMessage :: Maybe UTCTime -- ^ The time that message was added to the log.
```
This function is unnecessary, there is a `ToJSON` instance for `LogMessage`.

arybczak commented on log-base/src/Log/Backend/StandardOutput.hs:
```diff
@@ -20,6 +21,15 @@ withSimpleStdOutLogger act = do
+-- action, making sure that stdout is flushed afterwards.
+withSimpleJsonLineStdOutLogger :: (Logger -> IO r) -> IO r
+withSimpleJsonLineStdOutLogger act = do
+  logger <- mkLogger "jsonlines-stdout" $
```
Wouldn't bulk logger be better here? I'd imagine that getting bulk messages once every second is enough as they would be sent to ES or indexed somewhere anyway as opposed to read by people who want immediate feedback (but that's what simple stdout logger is for).

arybczak commented:
Superseded by #61.
